sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3064,13 +3064,31 @@ class Dataset[T] private[sql](
    */
   @scala.annotation.varargs
   def drop(col: Column, cols: Column*): DataFrame = {
-    val allColumns = col +: cols
-    val expressions = (for (col <- allColumns) yield col match {
+    val expressions = (col +: cols).map {
       case Column(u: UnresolvedAttribute) =>
-        queryExecution.analyzed.resolveQuoted(
```

Comment:
I think the problem is we put complicated resolution logic in DF APIs. Shall we add a new logical plan `DropColumns`? It will be rewritten to `Project` when the columns are all resolved, so that we can reuse all the column resolution logic we implemented in the analyzer.

Reply:
I think new plan `DropColumns` should be a better approach

Reply:
What's the behavior of
`dropped_1 = joined_df.select(left_df["value1"], right_df["join_key"], right_df["value2"])` ?